### PR TITLE
[python-0500] implement solution

### PR DESCRIPTION
**Pull request by CC972**

### CODE REVIEW

File: python/tests/test_problem_0500_keyboard_row.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+@pytest.mark.parametrize(
+    "words, expected",
+    [
+        pytest.param(["Hello", "Alaska", "Dad", "Peace"], ["Alaska", "Dad"]),
+        pytest.param(["omk"], []),
+        pytest.param(["adsdf", "sfd"], ["adsdf", "sfd"])
+    ]
+)
+def test_find_words_returns_words_which_can_be_typed_using_a_single_keyboard_row(words, expected):
```
Comment (GreenSlime96) [inline python/tests/test_problem_0500_keyboard_row.py:13]: where is the `given // when // then`? ;-)

File: python/solutions/problem_0500_keyboard_row.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    acceptable_words = []
+
+    first_row = set("qwertyuiop")
+    second_row = set("asdfghjkl")
+    third_row = set("zxcvbnm")
+
+    for word in words:
+
+        word_lower = set(word.lower())
```
Comment (GreenSlime96) [inline python/solutions/problem_0500_keyboard_row.py:16]: performance gain of converting to set is very negligible here.

if you use a set, you might as well use set subset checking (i.e `first_row <= word_lower or second_row <= word_lower or third_row <= word_lower`)

**Review by GreenSlime96 — CHANGES_REQUESTED** [review]

File: python/solutions/problem_0500_keyboard_row.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+
+from typing import List
+
+
+def find_words(words: List[str]) -> List[str]:
+    """Given a list of words, returns those which can be typed using a single row on the keyboard"""
+
+    acceptable_words = []
+
+    first_row = set("qwertyuiop")
```
Comment (GreenSlime96) [inline python/solutions/problem_0500_keyboard_row.py:10]: you can move this outside the function as it's static

**Review by GreenSlime96 — CHANGES_REQUESTED** [review]

File: python/solutions/problem_0500_keyboard_row.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    second_row = set("asdfghjkl")
+    third_row = set("zxcvbnm")
+
+    for word in words:
+
+        word_lower = set(word.lower())
+
+        for row in (first_row, second_row, third_row):
```
Comment (GreenSlime96) [inline python/solutions/problem_0500_keyboard_row.py:18]: if you are using sets of characters of `word_lower`, you can also do something like:

```
KEYBOARD_ROWS = [
    set("qwertyuiop"),
    set("asdfghjkl"),
    set("zxcvbnm")
]

if any(row <= word_lower for row in KEYBOARD_ROWS):
    acceptable_words.append(word)
```

Also, if you store capital letters inside the rows as well, you don't need to make passes to lowercase. However, this is a _micro_-optimisation you can ignore as converting to set can be quite expensive.

**Review by GreenSlime96 — CHANGES_REQUESTED** [review]

File: python/solutions/problem_0500_keyboard_row.py (hunk 2 above)
Comment (CC972) [reply python/solutions/problem_0500_keyboard_row.py:16]: Agreed that converting to a set is probably just a micro-optimisation. Also I think for subset checking it should be `word_lower <= first_row`, etc.

File: python/solutions/problem_0500_keyboard_row.py (hunk 4 above)
Comment (CC972) [reply python/solutions/problem_0500_keyboard_row.py:18]: Good idea as this eliminates the nested for loop :)

**Review by GreenSlime96 — APPROVED** [review]

**Pull request merged**